file_store/src/file_store.rs

```diff
@@ -31,20 +31,40 @@ impl FileStore {
             _ => None,
         };
         let region = Region::new(settings.region.clone());
-        Self::new(endpoint, region, &settings.bucket).await
+        Self::new(
+            endpoint,
+            region,
+            &settings.bucket,
+            settings.access_key_id.clone(),
+            settings.secret_access_key.clone(),
+        )
+        .await
     }
 
     pub async fn new(
         endpoint: Option<Endpoint>,
         region: impl ProvideRegion + 'static,
         bucket: impl Into<String>,
+        #[allow(unused_variables)] access_key_id: Option<String>,
+        #[allow(unused_variables)] secret_access_key: Option<String>,
```

Comment:
i'm guessing this whole thing would still work the same and these annotations would be unnecessary without the `local` feature cfg macro below but that's in place as an extra guard on top of just not supplying the `access_key_id` and `secret_access_key` as arguments outside of local testing?

Reply:
this is one of the arguments for the builder pattern.. it removes the need for these allow/feature hacks?

Reply:
I removed the `FileStore::new` function since it was only used in a test and moved the functionality into `FileStore::from_settings`.  This removes the need for the allow directives.

Address in 93b33e5

Reply:
nice fix